f5lbaasdriver/v2/bigip/test/test_service_builder.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+import pytest
+
+from f5lbaasdriver.v2.bigip.service_builder import LBaaSv2ServiceBuilder
+
+
+@pytest.fixture
+def listeners():
+    return [{'id': 'e6ce8fd6-907f-11e6-ae22-56b6b6499611'},
+            {'id': '218b4f6f-1243-494e-96a6-aba55759da69'}]
+
+
+@pytest.fixture
+def l7policies():
+    return [{'id': '2ea7511d-a911-484b-bf1a-8abc7b249d66'},
+            {'id': 'f5f4e752-e54e-45b8-a093-4c7587391855'}]
+
+
+@pytest.fixture
+def l7rules():
+    return [{'id': '850bb3cb-731d-4215-b345-0787a02a5be5'},
+            {'id': '45bb4ac2-df90-4ea6-a2fb-1ff50477a9d5'}]
+
+
+@pytest.fixture
+def service_builder():
+    return LBaaSv2ServiceBuilder(mock.MagicMock())
+
+
+def test_get_l7policies(listeners, l7policies):
+    """Test that get_l7policies returns valid list of dict"""
+    context = mock.MagicMock()
+    driver = mock.MagicMock()
+
+    # mock an L7 policy object with a to_dict() method
+    mock_policy = mock.MagicMock()
+    mock_policy.to_dict.return_value = l7policies[0]
+
+    service_builder = LBaaSv2ServiceBuilder(driver)
+    service_builder.driver.plugin.db.get_l7policies = mock.MagicMock(
+        return_value=[mock_policy])
+
+    policies = service_builder._get_l7policies(context, listeners)
+
```

Comment:
One good thing you can check is that the call made out to the driver.plugin is as you expect. Because it is a mock object, you can make an assertion about the list of calls made on that object with `assert service_builder.driver.plugin.db.get_l7policies.call_args_list == [mock.call(arg1, arg2)...]`

This makes for a little more white box type testing.